### PR TITLE
Attempting to clear some of the black magic in getSlots + new testcases

### PR DESCRIPTION
**Pull request by emrysal**
I set out to make the slot logic a bit more palatable, it had reached a point where to understand the logic, you just had to trust that it worked. Black magic 🙈 

That kind of code doesn't quite work longer term; so in this PR I tried to flatten the logic a bit, return earlier, decrease the timeZone logic & primarily iterate on the testing suite. I think it's starting to get to a point where one can look at the code and there's no need to be guided through it; so reviewing should be a lot easier now.

Feedback welcome :)

### CODE REVIEW

**Review by pumfleet — APPROVED** [review]
As far as I can see, it looks good. I'm no expert on availability logic though, so will wait for others' approval before merging :)

**Comment by joshsny** [comment]
@emrysal Am I being silly, or is the users schedule setup in the onboarding flow not used in this (and the previous) availability logic?

**Review by zomars — APPROVED** [review]
Looking good!

File: lib/slots.ts
Comment (alishaz-polymath) [inline lib/slots.ts:14]: I'd rather go 
`value === 0 ? 0 : cb(value / frequency) * frequency` because it appears more linear this way, but that's a subjective thing in the end.

**Comment by alishaz-polymath** [comment]
Hey @emrysal. Everything looks good, but when I ran a test case, the result was off. Here's a loom video showcasing it... I'm not entirely sure why this is, but perhaps you can test it out on your end as well, just in case, before I approve this. 
https://www.loom.com/share/37661a7240204ea2aaa3d61b80f5f712

**Comment by alishaz-polymath** [comment]
> 
> 
> Hey @emrysal. Everything looks good, but when I ran a test case, the result was off. Here's a loom video showcasing it... I'm not entirely sure why this is, but perhaps you can test it out on your end as well, just in case, before I approve this.
> https://www.loom.com/share/37661a7240204ea2aaa3d61b80f5f712

Even in other timezone test (Asia/Kolkata), the time slots are off. Like, really off... I think this needs more testing for sure!

**Review by KATT — COMMENTED** [review]
Looks like an improvement to me! Does all the test now work (i.e. can #649 be merged after this?)

**Comment by PeerRich** [comment]
shit, we didnt merged this in time, now there're new conflicts. @emrysal can you take a look at the conflicting files and @baileypumfleet can you merge it in after? 🙏

**Review by PeerRich — APPROVED** [review]

**Review by PeerRich — APPROVED** [review]

**Review by pumfleet — APPROVED** [review]
Really cool work with the slot tests :)

**Comment by emrysal** [comment]
Outdated due to availability revamp

**Pull request closed without merging**
